src/RedHat.AspNetCore.Server.Kestrel.Transport.Linux/Transport.cs

```diff
@@ -141,11 +134,30 @@ public async Task BindAsync()
             }
             catch
             {
-                await StopAsync();
+                await DisposeAsync();
                 throw;
             }
         }
 
+        public async ValueTask<ConnectionContext> AcceptAsync(CancellationToken cancellationToken = default)
+        {
+            lock (_gate)
+            {
+                if (_state >= State.Stopping)
```

Comment:
Can you change this to:

`state >= Unbinding` -> `return null`
`state != Bound` -> `ThrowInvalidOperation();`

Reply:
This doesn't feel right to me.

If Transport.AcceptAsync() starts returning null as soon as it transitions into the Unbinding state, that means that there could be calls to ThreadContext.AcceptAsync() from Transport.AcceptConnnections() that are still pending and will never get observed if they complete. If one of those pending calls were to return a ConnectionContext instead of null, that ConnectionContext would never be handled or aborted.

Reply:
Ah yes, as long as we're Unbinding, the caller should be calling AcceptAsync.